app/assets/javascripts/modules/MultiTableFilter.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * # Multiple table filter
+ *
+ * Filters across one or more tables, and can also hide the elements associated with a table.
+ *
+ * Requires an element to have a data-dough-component="MultiTableFilter" attribute and an ID.
+ *
+ * Makes use of jquery.fastLiveFilter.
+ * See test fixture for sample markup - /spec/js/fixtures/MultiTableFilter.html
+ */
+
+define(['jquery', 'jqueryFastLiveFilter', 'DoughBaseComponent'],
+       function($, jqueryFastLiveFilter, DoughBaseComponent) {
+  'use strict';
+
+  var MultiTableFilterProto,
+      defaultConfig = {
+        filterTargetSelector: '.js-filterable',
+        filterFieldSelector: '.js-filter-input',
+        filterListSelector: '.js-filter-rows',
+        filterWrapperSelector: '.js-filter-group'
+      };
+
+  function MultiTableFilter($el, config) {
+    MultiTableFilter.baseConstructor.call(this, $el, config, defaultConfig);
+  }
+
+  /**
+   * Inherit from base module, for shared methods and interface
+   * @type {[type]}
+   * @private
+   */
+  DoughBaseComponent.extend(MultiTableFilter);
+  MultiTableFilterProto = MultiTableFilter.prototype;
+
+  /**
+   * Init function
+   *
+   * Set up listeners and accept promise
+   *
+   * @param  {[type]} initialised [description]
+   * @return {[type]}             [description]
+   */
+  MultiTableFilterProto.init = function(initialised) {
+    this.bindFilterToElements();
+    this._initialisedSuccess(initialised);
+    return this;
+  };
+
+  /**
+   * show/hide filterWrappers when all filterTargetSelectores are hidden
+   */
+  MultiTableFilterProto.toggleFilterWrappers = function() {
+    var self = this;
+
+    $(self.config.filterWrapperSelector).each(function(_index, filterWrapper) {
+      self.toggleFilterWrapper(filterWrapper);
+    });
+  };
+
+  /**
+   * show/hide filterWrapper when all filterTargetSelectores are hidden
+   */
+
+  MultiTableFilterProto.toggleFilterWrapper = function(filterWrapper) {
+    var $filterWrapper = $(filterWrapper),
+        $list = $filterWrapper.find(this.config.filterListSelector),
+        $rows = $list.first().find('tr'),
+        hiddenRows;
+
+    hiddenRows = $rows.filter(function() {
+      return $(this).css('display') == 'none';
```

Comment:
Could try the [`:hidden`](https://api.jquery.com/hidden-selector/) filter here – what do you think for readability?

```js
return $(this).is(':hidden');
```